<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Tjeerd Hoogendijk</author>
 <class>MechaduinoMainWindow</class>
 <widget class="QMainWindow" name="MechaduinoMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1600</width>
    <height>900</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1600</width>
    <height>900</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1600</width>
    <height>900</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Mechaduino tuning</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidgetMain">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="ComSettings">
       <attribute name="title">
        <string>Communication settings</string>
       </attribute>
       <widget class="QLabel" name="label_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>7</y>
          <width>929</width>
          <height>17</height>
         </rect>
        </property>
        <property name="text">
         <string>Serial port mechaduino</string>
        </property>
       </widget>
       <widget class="QComboBox" name="comboBoxSerialPort">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>929</width>
          <height>33</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="ConnectMechaduino">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>70</y>
          <width>85</width>
          <height>27</height>
         </rect>
        </property>
        <property name="text">
         <string>Connect</string>
        </property>
       </widget>
       <widget class="QTextBrowser" name="textBrowserComStatus">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>110</y>
          <width>929</width>
          <height>551</height>
         </rect>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Step response</string>
       </attribute>
       <widget class="PlotWidget" name="GraphTorque">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>1100</width>
          <height>200</height>
         </rect>
        </property>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>1120</x>
          <y>340</y>
          <width>391</width>
          <height>184</height>
         </rect>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <property name="spacing">
          <number>4</number>
         </property>
         <item row="0" column="2">
          <widget class="QPushButton" name="setGainspPID">
           <property name="text">
            <string>set p PID</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Integral gain</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QDoubleSpinBox" name="pKpGain">
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="value">
            <double>30.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Proportional gain</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="setGainsvPID">
           <property name="text">
            <string>set v PID</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="vKpGain">
           <property name="value">
            <double>0.800000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Derivetive gain</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Lowpass cut-off</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QDoubleSpinBox" name="pKiGain">
           <property name="maximum">
            <double>10.000000000000000</double>
           </property>
           <property name="value">
            <double>0.010000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QDoubleSpinBox" name="pKdGain">
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="value">
            <double>500.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="vKiGain">
           <property name="value">
            <double>0.010000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="vKdGain">
           <property name="decimals">
            <number>3</number>
           </property>
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="value">
            <double>-1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="4" column="2">
          <widget class="QDoubleSpinBox" name="doubleSpinBox_14"/>
         </item>
         <item row="4" column="1">
          <widget class="QDoubleSpinBox" name="doubleSpinBox_13"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>1120</x>
          <y>170</y>
          <width>391</width>
          <height>157</height>
         </rect>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <property name="spacing">
          <number>5</number>
         </property>
         <item row="0" column="3">
          <widget class="QPushButton" name="do_pStep">
           <property name="text">
            <string>pStep</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Amplitude</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="do_vStep">
           <property name="text">
            <string>vStep</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="tStepAmplitude">
           <property name="minimum">
            <double>-2000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>2000.000000000000000</double>
           </property>
           <property name="value">
            <double>1000.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Start step (ms)</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QDoubleSpinBox" name="pStepAmplitude">
           <property name="minimum">
            <double>-36000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>36000.000000000000000</double>
           </property>
           <property name="value">
            <double>90.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="tStepEnd">
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>50</number>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="do_tStep">
           <property name="text">
            <string>tStep</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="tStepStart">
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QDoubleSpinBox" name="vStepAmplitude">
           <property name="minimum">
            <double>-1000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="value">
            <double>100.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>End step (ms)</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QSpinBox" name="vStepEnd">
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>50</number>
           </property>
          </widget>
         </item>
         <item row="3" column="3">
          <widget class="QSpinBox" name="pStepEnd">
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>50</number>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QSpinBox" name="vStepStart">
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QSpinBox" name="pStepStart">
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QTextBrowser" name="textBrowserStepInfo">
        <property name="geometry">
         <rect>
          <x>1120</x>
          <y>10</y>
          <width>391</width>
          <height>151</height>
         </rect>
        </property>
       </widget>
       <widget class="PlotWidget" name="GraphVelocity">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>200</y>
          <width>1100</width>
          <height>200</height>
         </rect>
        </property>
       </widget>
       <widget class="PlotWidget" name="GraphPosition">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>400</y>
          <width>1100</width>
          <height>200</height>
         </rect>
        </property>
       </widget>
       <widget class="PlotWidget" name="GraphError">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>600</y>
          <width>1100</width>
          <height>200</height>
         </rect>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1600</width>
     <height>27</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionDf">
   <property name="text">
    <string>df</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionLog">
   <property name="text">
    <string>Log</string>
   </property>
  </action>
  <action name="actionLog_2">
   <property name="text">
    <string>Log</string>
   </property>
  </action>
  <action name="actionInition_table">
   <property name="text">
    <string>Inition table</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save as</string>
   </property>
  </action>
  <action name="actionOpen_project">
   <property name="text">
    <string>Open log</string>
   </property>
  </action>
  <action name="actionNew_project">
   <property name="text">
    <string>New log</string>
   </property>
  </action>
  <action name="actionClose_project">
   <property name="text">
    <string>Close log</string>
   </property>
  </action>
  <action name="actionLoad_tune">
   <property name="text">
    <string>Load log</string>
   </property>
  </action>
  <action name="actionSave_tune">
   <property name="text">
    <string>Save settings</string>
   </property>
  </action>
  <action name="actionSave_Tune_as">
   <property name="text">
    <string>Save log as</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionStart_Logging">
   <property name="text">
    <string>Start Logging</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="text">
    <string>Stop</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionSafe_Project">
   <property name="text">
    <string>Safe log</string>
   </property>
  </action>
  <action name="actionSave_Project_as">
   <property name="text">
    <string>Save log as</string>
   </property>
  </action>
  <action name="actionOpen_settings">
   <property name="text">
    <string>Open settings</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
